### PR TITLE
Aspect defaultDescription & typeDescription

### PR DESCRIPTION
**Pull request by valentinyanakiev**

### CODE REVIEW

**Review by techsmyth — CHANGES_REQUESTED** [review]
Overall functionality looks good. However some confusion related to the renaming:
- on the Aspect entity the field should be called "description"
- on the AspectTemplate entity the field should be called "defaultDescription"

And then when the aspect is being created the client should pick up the defaultDescription field from the AspectTemplate to fill in the initial value of the description box.

File: graphql-samples/mutations/create/create-aspect-on-context [hunk 1]
```diff
@@ -1,7 +1,8 @@
 mutation CreateAspect($aspectData: CreateAspectInput!) {
   createAspectOnContext(aspectData: $aspectData) {
     displayName,
-    description,
+    defaultDescription,
```
Comment (techsmyth) [inline graphql-samples/mutations/create/create-aspect-on-context]: this should be description, defaultDescription is only in the template

File: graphql-samples/mutations/create/create-aspect-on-context [hunk 2]
```diff
@@ -14,7 +15,8 @@ query variables:
     "contextID": "uuid",
     "type": "test",
     "displayName": "TechStack2",
-    "description": "How do I handle scalability?",
+    "defaultDescription": "How do I handle scalability?",
```
Comment (techsmyth) [inline graphql-samples/mutations/create/create-aspect-on-context]: ditto

File: src/domain/context/aspect/dto/aspect.dto.update.ts [hunk 3]
```diff
@@ -10,6 +10,10 @@ export class UpdateAspectInput extends UpdateNameableInput {
   @MaxLength(LONG_TEXT_LENGTH)
   description?: string;
 
+  @Field({ nullable: true })
```
Comment (techsmyth) [inline src/domain/context/aspect/dto/aspect.dto.update.ts:13]: No need to update the type description; also impacts the updateAspect method

File: src/domain/context/aspect/aspect.service.ts [hunk 4]
```diff
@@ -120,6 +120,10 @@ export class AspectService {
       aspect.description = aspectData.description;
     }
 
+    if (aspectData.typeDescription) {
```
Comment (techsmyth) [inline src/domain/context/aspect/aspect.service.ts]: Not sure why we would need to update the typeDescription, it is fixed - and also questionable if it should be stored per entity - but lets not pick this up now.

File: src/domain/challenge/hub/dto/hub.dto.template.aspect.ts [hunk 5]
```diff
@@ -6,10 +6,18 @@ export class AspectTemplate {
   type: string;
 
   @Field(() => String)
-  description: string;
+  defaultDescription: string;
 
-  constructor(type: string, description?: string) {
+  @Field(() => String)
+  typeDescription: string;
```
Comment (techsmyth) [inline src/domain/challenge/hub/dto/hub.dto.template.aspect.ts:18]: Good to add in the description metadata for the graphql api

File: src/domain/challenge/hub/dto/hub.dto.template.aspect.ts [hunk 6]
```diff
@@ -6,10 +6,18 @@ export class AspectTemplate {
   type: string;
 
   @Field(() => String)
-  description: string;
+  defaultDescription: string;
```
Comment (techsmyth) [inline src/domain/challenge/hub/dto/hub.dto.template.aspect.ts:13]: Good to add in the description metadata for the graphql api

**Review by techsmyth — CHANGES_REQUESTED** [review]
Mostly there, still some minor updates.

File: src/domain/context/aspect/dto/aspect.dto.update.ts (hunk 3 above)
Comment (valentinyanakiev) [reply src/domain/context/aspect/dto/aspect.dto.update.ts:13]: So if somebody adds a type description and e.g. copy / pastes some description incorrectly, the option to correct it won't exist? Is that really what we want?

File: src/domain/context/aspect/aspect.service.ts (hunk 4 above)
Comment (valentinyanakiev) [reply src/domain/context/aspect/aspect.service.ts]: I agree that it's not ideal to be per entity, but that will open another set of updates, similarly to preference / preferenceDefinition.

**Review by techsmyth — APPROVED** [review]

**Pull request merged**